### PR TITLE
Move to statically linked zlib

### PR DESCRIPTION
**Pull request by JanuszL**
- to avoid collision with system zlib symbols and make
  DALI to use own zlib version (zlib-ng) moves to static version of zlib
- excludes all zlib symbols from libtiff to avoid overlap with system zlib

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

**Review by stiepan — APPROVED** [review]

**Review by mzient — APPROVED** [review]

File: build_scripts/build_opencv.sh [hunk 1]
```diff
@@ -23,6 +23,7 @@ cmake -DCMAKE_BUILD_TYPE=RELEASE \
       -DCMAKE_TOOLCHAIN_FILE=$PWD/../platforms/${OPENCV_TOOLCHAIN_FILE} \
       -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
       -DBUILD_LIST=core,improc,imgcodecs \
+      -DZLIB_ROOT=${INSTALL_PREFIX} \
```
Comment (mzient) [inline build_scripts/build_opencv.sh:26]: I think we should have a separate variable just for that and only use this as a default.
```suggestion
      -DZLIB_ROOT=${ZLIB_ROOT:-$INSTALL_PREFIX} \
```

Comment (JanuszL) [reply build_scripts/build_opencv.sh:26]: Do you think we should update this inside https://github.com/NVIDIA/DALI_deps/blob/main/build_scripts/build_libtiff.sh#L30 as well?

Comment (mzient) [reply build_scripts/build_opencv.sh:26]: Probably....

Comment (JanuszL) [reply build_scripts/build_opencv.sh]: Done

**Review by mzient — APPROVED** [review]

**Review by mzient — APPROVED** [review]

**Review by stiepan — APPROVED** [review]

**Pull request merged**